### PR TITLE
Run nightly promotion build once per week for old release branches

### PR DESCRIPTION
**Pull request by blindpirate**
For old release branches, we still want to run nightly promotion job infrequently, like once per week.

### CODE REVIEW

File: .teamcity/src/test/kotlin/VersionedSettingsBranchTest.kt [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        ]
+    )
+    fun validBranches(branchName: String, expectedEnableVcsTriggers: Boolean, expectedNightlyPromotionTriggerHour: Int?) {
+        val vsb = VersionedSettingsBranch(branchName)
+        assertEquals(expectedEnableVcsTriggers, vsb.enableVcsTriggers)
+        assertEquals(expectedNightlyPromotionTriggerHour, vsb.nightlyPromotionTriggerHour)
+    }
+
+    @Test
+    fun invalidBranches() {
```
Comment (wolfs) [inline .teamcity/src/test/kotlin/VersionedSettingsBranchTest.kt:42]: ❌ Can you also add a test for a branch which doesn't match the `release...` pattern?

File: .teamcity/src/main/kotlin/common/VersionedSettingsBranch.kt [hunk 2]
```diff
@@ -2,7 +2,28 @@ package common
...
-data class VersionedSettingsBranch(val branchName: String, val enableTriggers: Boolean) {
+data class VersionedSettingsBranch(val branchName: String) {
+    /**
+     * Whether the <a href="https://www.jetbrains.com/help/teamcity/configuring-vcs-triggers.html">VCS trigger</a>
+     * should be enabled, i.e. when new commits are pushed to this branch, should a ReadyForNightly job
+     * be triggered automatically?
+     *
+     * Currently, we only enable VCS trigger for `master` and `release` branches.
+     */
+    val enableVcsTriggers: Boolean = mainBranches.contains(branchName.lowercase())
```
Comment (wolfs) [inline .teamcity/src/main/kotlin/common/VersionedSettingsBranch.kt:13]: ❓ Why shouldn't we enable VCS triggers on the `release...` branches? I suppose we shouldn't enable VCS triggers on the `experimental` branch, though for the old release branches it seems to make sense to run a build when there are changes.

Comment (blindpirate) [reply .teamcity/src/main/kotlin/common/VersionedSettingsBranch.kt:13]: That's how the code works before. But yes, it makes sense to run them for old release branches.

**Review by wolfs — APPROVED** [review]

**Comment by ljacomet** [comment]
From the description and comments from Stefan:
* Running builds and nightly promotion when there are changes makes sense
* Running a Ready for Nightly once in a while makes sense to make sure we do not regress in CI setup for older branches

I am not sure I fully understand the changes here, but if it does the above then ✅

**Pull request merged**